Framework/DataHandling/src/LoadDetectorInfo.cpp

```diff
@@ -187,16 +190,17 @@ void LoadDetectorInfo::loadFromRAW(const std::string &filename) {
 
   // Start loop over detectors
   auto &pmap = m_workspace->instrumentParameters();
+  const auto &detectorInfo = m_workspace->detectorInfo();
   for (int i = 0; i < numDets; ++i) {
     detid_t detID = static_cast<detid_t>(iraw.udet[i]);
     int code = iraw.code[i];
-    IDetector_const_sptr det;
+    size_t index;
     try {
-      det = m_baseInstrument->getDetector(detID);
-    } catch (Exception::NotFoundError &) {
+      index = detectorInfo.indexOf(detID);
+    } catch (std::out_of_range &) {
```

Comment:
`API::DetectorInfo::indexOf` throws `std::out_of_range`. Also, it will be removed once we deal with all the indexing things.